tests/integration/admin_test.py

```diff
@@ -80,11 +84,161 @@ def test_update_groups(client: AtlanClient, group1: CreateGroupResponse):
     client.update_group(group)
 
 
-# TODO: users and then group membership
+def _get_user1_from_list(users: list[AtlanUser]) -> Optional[AtlanUser]:
+    for user in users:
+        if user.email == USER_EMAIL1.lower():
+            return user
+    return None
+
+
+def _get_user2_from_list(users: list[AtlanUser]) -> Optional[AtlanUser]:
+    for user in users:
+        if user.email == USER_EMAIL2.lower():
+            return user
+    return None
+
+
+def _get_user3_from_list(users: list[AtlanUser]) -> Optional[AtlanUser]:
+    for user in users:
+        if user.email == USER_EMAIL3.lower():
+            return user
+    return None
+
+
+@pytest.fixture(scope="module")
+def users(client: AtlanClient) -> Generator[list[AtlanUser], None, None]:
+    users: list[AtlanUser] = [
+        AtlanUser.create(email=USER_EMAIL1, role_name="$guest"),
+        AtlanUser.create(email=USER_EMAIL2, role_name="$guest"),
+        AtlanUser.create(email=USER_EMAIL3, role_name="$guest"),
+    ]
+    client.create_users(users=users)
+    created = client.get_users_by_email(email=EMAIL_DOMAIN)
+    assert created  # must be here for mypy
+    yield created
+    for user in created:
+        client.purge_user(str(user.id))
+
+
+@pytest.mark.order(after="test_retrieve_all_groups")
+def test_retrieve_users(client: AtlanClient, users: list[AtlanUser]):
+    global _default_group_count
```

Comment:
I don't, either, but this was such a one-off I didn't think it worthwhile doing an entire refactoring (especially as it's just a test rather than core SDK code).